vpd-manager/src/parser.cpp

```diff
@@ -56,111 +58,144 @@ types::VPDMapVariant Parser::parse()
+            types::ObjectMap l_dbusObjMap = {std::make_pair(
+                l_inventoryObjPath,
+                types::InterfaceMap{std::make_pair(
+                    l_interfaceName, types::PropertyMap{std::make_pair(
+                                         l_propertyName, l_keywordValue)})})};
 
-        // Call PIM's Notify method to perform update
-        if (!dbusUtility::callPIM(std::move(l_dbusObjMap)))
+            // Call PIM's Notify method to perform update
+            if (!dbusUtility::callPIM(std::move(l_dbusObjMap)))
+            {
+                // Call to PIM's Notify method failed.
+                std::string l_errMsg("Notify PIM is failed for object path: " +
+                                     l_inventoryObjPath);
+                throw std::runtime_error(l_errMsg);
+            }
+        }
+
+        // Update keyword's value on redundant hardware if present
+        if (!l_redundantFruPath.empty())
         {
-            // Call to PIM's Notify method failed.
-            logging::logMessage("Notify PIM is failed for object path: " +
-                                l_inventoryObjPath);
-            return -1;
+            if (updateVpdKeywordOnRedundantPath(l_redundantFruPath,
+                                                i_paramsToWriteData) < 0)
+            {
+                std::string l_errMsg(
+                    "Error while updating keyword's value on redundant path " +
+                    l_redundantFruPath);
+                throw std::runtime_error(l_errMsg);
+            }
         }
-    }
 
-    // Update keyword's value on redundant hardware if present
-    if (!l_redundantFruPath.empty())
+        // TODO: Check if revert is required when any of the writes fails.
+        // TODO: Handle error logging
+    }
+    catch (const std::exception& l_ex)
     {
-        if (updateVpdKeywordOnRedundantPath(l_redundantFruPath,
-                                            i_paramsToWriteData) < 0)
+        std::string l_keywordIdentifier{};
+        if (const types::IpzData* l_ipzData =
+                std::get_if<types::IpzData>(&i_paramsToWriteData))
         {
-            logging::logMessage(
-                "Error while updating keyword's value on redundant path " +
-                l_redundantFruPath);
-            return -1;
+            l_keywordIdentifier = std::get<0>(*l_ipzData) + ":" +
+                                  std::get<1>(*l_ipzData);
         }
-    }
+        else if (const types::KwData* l_kwData =
+                     std::get_if<types::KwData>(&i_paramsToWriteData))
+        {
+            l_keywordIdentifier = std::get<0>(*l_kwData);
+        }
+        logging::logMessage(
+            "Update VPD Keyword failed for : " + l_keywordIdentifier +
+            " failed due to error: " + l_ex.what());
 
-    // TODO: Check if revert is required when any of the writes fails.
-    // TODO: Handle error logging
+        return constants::FAILURE;
```

Comment:
we dont need this statement, as we will return at the end anyways. ?

Reply:
Suppose line 69 `writeKeywordOnHardware` is successful, `l_bytesUpdatedOnHardware` has some value > 0,  but any subsequent steps like `updateVpdKeywordOnRedundantPath` fails, we have to return -1.

Reply:
Addressed.